File: 2_PCs/Ponto de controle 2/abre_cancela.py
import RPi.GPIO as GPIO
import time
import signal

GPIO.setmode(GPIO.BCM)

## Definicao dos pinos de saida
STEP = 16
DIR = 26

def setup():
        print("Setando a GPIO...")
        GPIO.setup(STEP,GPIO.OUT)
        GPIO.setup(DIR,GPIO.OUT)
        time.sleep(0.5)
def close():
        GPIO.cleanup()
        print("Fechando programa...")
        time.sleep(0.5)
        exit()


def driver(ligar, direcao):

        if(ligar == 1):
                print("Ligado")
                if(direcao == 1):
                        print("Direita")
                        GPIO.output(DIR, GPIO.HIGH)
                        GPIO.output(STEP, GPIO.HIGH)
                        time.sleep(0.05)
                        GPIO.output(STEP, GPIO.LOW)
                        time.sleep(0.05)

                if(direcao == 0):
                        print("Esquerda")
                        GPIO.output(DIR, GPIO.LOW)
                        GPIO.output(STEP, GPIO.HIGH)
                        time.sleep(0.05)
                        GPIO.output(STEP, GPIO.LOW)
                        time.sleep(0.05)  

        if(ligar == 0):
                print("Desligado")
                GPIO.output(STEP, GPIO.LOW)
                GPIO.output(DIR, GPIO.LOW)


if __name__ == "__main__":
        try:
                setup()
                while(1):
                        Abre = 2
                        Abre = input("Abre: (1) || Fecha: (0) - ")
                        if(Abre == 1):
                                for x in range(1,50):
                                        driver(1,1)
                        else:
                                for x in range(1,50):
                                        driver(1,0)


        # Cancelar  CTRL + C
        except KeyboardInterrupt:
                close()

File: 2_PCs/Ponto de Controle 3/abre_cancela.c
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>
#include <time.h>
#include <unistd.h>
#include <fcntl.h>
#include <wiringPi.h>

#define STEP 27
#define DIR 25

void ctrl_c()
{
        printf("\n Fechando o programa! \n");
        exit(0);
}

void driver(int control, int passos)
{
        int i =0;
        if(control == 1)
        {
                for(i = 0; i < passos; i++)
                {
                        digitalWrite(DIR, 1);
                        digitalWrite(STEP, 1);
                        delay(1);
                        digitalWrite(STEP, 0);
                        delay(1);
                }
        }
        else
        {
                for(i = 0; i < passos; i++)
                {
                        digitalWrite(DIR, 0);
                        digitalWrite(STEP, 1);
                        delay(1);
                        digitalWrite(STEP, 0);
                        delay(1);
                }
        }

}
int main()
{
        int control = -1;
        int passos = 50;

        signal(SIGINT, ctrl_c);

        wiringPiSetup();
        pinMode(STEP, OUTPUT);
        pinMode(DIR, OUTPUT);

        while(1)
        {
                control = -1;
                printf("Abre (1) e Fecha (0): ");
                scanf("%d", &control);
                printf("%d \n", control);
                driver(control,passos);
        }

        return 0;
}


